test/spec/browser.js

```diff
@@ -93,22 +101,92 @@ describe('Browser', function() {
     });
   });
 
+  describe('signIn', () => {
+    describe('loginWithRedirect', () => {
+      beforeEach(() => {
+        auth.loginRedirect = jest.fn();
+        auth.setFromUri = jest.fn();
+      });
+      it('calls loginRedirect by default', async () => {
+        expect.assertions(1);
+        await auth.signIn();
+        expect(auth.loginRedirect).toHaveBeenCalled();
+      });
+  
+      it('calls onAuthRequired, if provided, instead of loginRedirect', async () => {
+        auth.options.onAuthRequired = jest.fn().mockResolvedValue();
+        await auth.signIn();
+        expect(auth.loginRedirect).not.toHaveBeenCalled();
+        expect(auth.options.onAuthRequired).toHaveBeenCalledWith(auth);
+      });
+  
+      it('Calls setFromUri with fromUri, if provided', () => {
+        const fromUri = 'notrandom';
+        auth.signIn({ fromUri });
+        expect(auth.setFromUri).toHaveBeenCalledWith(fromUri);
+      });
+  
+      it('Calls setFromUri with undefined, by default', () => {
```

Comment:
Really not sure about this. Did this come from react? I don't think we want to call setFromUri unless a fromURI has been passed. The user may have called setFromUri explicitly and we would not want to overwrite the value with undefined.